### PR TITLE
Update navigation on empty feeds

### PR DESCRIPTION
**Pull request by aledeg**
When using feed navigation, the previous behavior was to cycle through all
available feeds regardless of their content. To match the behaviour of the
first feed and last feed navigation shortcuts, the navigation now skips
empty feeds. Now it's consistent through out the application.
When using feed navigation with only empty feeds, the new behavior is to
cycle through all available feeds.

See #2385

### CODE REVIEW

File: p/scripts/main.js [hunk 1]
```diff
@@ -464,11 +464,16 @@ function prev_feed() {
...
 		const feed = feeds[i];
-		if (found && getComputedStyle(feed).display !== 'none') {
+		if (feed.classList.contains('active')) {
+			found = true;
+			continue;
+		}
+		if (!found) {
+			continue;
+		}
+		if (feed.dataset.unread != 0 && getComputedStyle(feed).display !== 'none') {
```
Comment (Frenzie) [inline p/scripts/main.js]: I'm only taking a quick glance, but is this even reachable? The first if statement sets found to true and immediately continues to the next iteration, while the one directly following continues if found isn't true?

Comment (aledeg) [reply p/scripts/main.js]: This is reachable. I've tested it with many different configurations and they all worked.
The first if statement catches the current feed. The second if statement is there to not process feeds until we found the current feed. The third if statement does the job of switching feeds.

I hope this makes sense!

**Review by Alkarex — APPROVED** [review]

**Review by marienfressinaud — COMMENTED** [review]
The shortcuts don't work anymore if there is no unread articles, isn't it? So we can't navigate when all articles are read

**Comment by aledeg** [comment]
@marienfressinaud that's right. We can add something to fix that. I don't know if it's useful and/or needed.

**Comment by marienfressinaud** [comment]
I never use these shortcuts so I may not be the best person to take this decision, but I think it's strange to not allow navigation in this situation. Also, it would be not consistent with categories and entries navigation.

**Comment by aledeg** [comment]
@marienfressinaud now, shortcuts cycle through all feeds when all feeds are read.

**Review by marienfressinaud — APPROVED** [review]
I'm ok with the behaviour.

(nitpick) About the code, I think you can refactor a bit by using another variable than `adjacent` (I would call it `next_feed` or `prev_feed` but it's already the names of the functions). A simplified algo is:

```js
// ...
for(...) {
    // ...
    if (getComputedStyle(feed).display === 'none') { 
        continue
    }

    if (feed.unread !== 0 || next_feed === null) {
        next_feed = feed
    }
}

if (next_feed) {
    delayedClick(next_feed)
} else {
    last_feed()
}
```

You also can distinguish unread test from the null test to benefit of the early return in the first case.

**Comment by aledeg** [comment]
@marienfressinaud I've made some changes but there is something that's not working in what you mentioned. I need a way of separating the next feed (read or not) from the next not read feed.

> ```js
> if (feed.unread !== 0 || next_feed === null) {
>     next_feed = feed
> }
> ```

I think this wont

**Comment by Alkarex** [comment]
@aledeg I have not understood your last comment :-) Do you want to make additional changes?

**Comment by aledeg** [comment]
@Alkarex I want pointing out that the snippet that @marienfressinaud wrote was probably broken. I took into consideration some of the comments and refactor the code. I think this PR is complete. :)
